### DISCUSSION TITLE
Using graph.findModelsUnderElement(cell) with lables positioned outside the rectangle cell

### DISCUSSION THREAD

**Discussion by ROYopedia** [clientIO/joint#2406] [Q&A]
### What happened?

Hello,
we are using jointjs+ and I have one scenario where while dragging I am validating where the cell should avoid overlapping. 
I have taken code reference from validators.js  from planogram demo app of yours.

Now it is working fine in general with below code
```
export const doesShelfOverlap = (graph, cell) => {
    return graph.findModelsUnderElement(cell).length > 0;
};
```

But I have some labels which is positioned outside of my custom element. In this case the graph.findModelsUnderElement(cell).length > 0; is not able to detect overlapping models under it as only labels touching the other cells. 
Please refer to the below image where label of second rect is overlapping to the first rectangle.
![image](https://github.com/clientIO/joint/assets/22453886/4b84b167-fc8b-4a51-ad45-aac01d2057ef)

Any suggestion you can give to handle the situation.

### Version

3.7

### What browsers are you seeing the problem on?

Firefox, Chrome, Android (State browser in description)

### What operating system are you seeing the problem on?

Linux

**Comment by kumilingus** [comment]
You need to work either with views (use [Paper.prototype.findViewsInArea](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Paper.prototype.findViewsInArea)) or write a custom function `findModelsUnderElement` that adds a padding to the top of the model bounding box.

**Reply by ROYopedia** [reply]
A little unclear.
Any example you want to provide?

**Reply by kumilingus** [reply]
Sure.

Using views instead of models:
```js
function findViewsUnderElement(paper, elementView) {
    const views = [];
    paper.findViewsInArea(elementView.getBBox()).reduce((views, view) => {
        if (view.model === elementView.model) return views;
        if (view.model.isElement()) views.push(view);
        return views;
    }, views);
    return views;
}
```

Extend the element bounding boxes to contain the labels:
```js
function getExtendedBBox(el) {
    return el.getBBox().moveAndExpand({ x: 0, y: -40, width: 40, height: 0 });
}

function findModelsUnderElement(graph, element) {
    const models = [];
    const bbox = getExtendedBBox(element);
    graph.getElements().reduce((models, curEl) => {
        if (curEl === element) return models;
        const curElBBox = getExtendedBBox(curEl);
        if (curElBBox.intersect(bbox)) models.push(curEl);
        return models;
    }, models);
    return models;
}
```
